### PR TITLE
[CALCITE-2442] Remove .toDelete cassandra temp folder on Windows after tests

### PR DESCRIPTION
**Pull request by asereda-gs**
Several committers complained about stale directory `.toDelete` on Windows which make some release validation tests fail.

### CODE REVIEW

**Comment by DonnyZone** [comment]
This fix still fails to delete the `.toDelete` file. Are there any approach to figure out which thread lock the file?

**Comment by asereda-gs** [comment]
@DonnyZone the issue is that cassandra daemon is not properly shutdown after tests so some files are locked by cassandra service.

I have migrated Cassandra tests to JUnit5 which allows better control of resources (before/after callbacks). What I need is a callback after *all* tests have been executed. 

Testing now

**Comment by vlsi** [comment]
> What I need is a callback after all tests have been executed

I don't think JUnit has a callback "after all tests".
So there are multiple solutions:

a) use TestContainers
b) use "after each test class" callback


A workaround might be to use a single class for all the tests, then "afterclass" would be the same ase "after all cassandra tests".

**Comment by asereda-gs** [comment]
I have registered CassandraResource (which implements [CloseableResource](https://junit.org/junit5/docs/5.1.0/api/org/junit/jupiter/api/extension/ExtensionContext.Store.CloseableResource.html)) in root context.

It seems to work.

**Comment by asereda-gs** [comment]
@DonnyZone  can you pls check now (unfortunately I don't have easy access to Windows OS runtime) ?

**Comment by vlsi** [comment]
@asereda-gs , can you please remove this line https://github.com/apache/calcite/blob/10a5b8a89d319e6fed563e7e49518cfc960b93d6/cassandra/gradle.properties#L20 so junit4 is not added to cassandra/../test classpath?

File: cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+    /**
+     * Extract {@link CassandraDaemon} instance using reflection. It will be used
+     * to shutdown the cluster
+     */
+    private static CassandraDaemon extractDaemon() {
+      try {
+        Field field = EmbeddedCassandraServerHelper.class.getDeclaredField("cassandraDaemon");
+        field.setAccessible(true);
+        CassandraDaemon daemon = (CassandraDaemon) field.get(null);
```
Comment (vlsi) [inline cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java]: Is this really required?
I thought it should have a way to properly shutdown cassandra

File: cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+   * @return {@code true} if test is compatible with current environment,
+   *         {@code false} otherwise
+   */
+  @Override public ConditionEvaluationResult evaluateExecutionCondition(
+      final ExtensionContext context) {
+    boolean enabled = CalciteSystemProperty.TEST_CASSANDRA.value();
+    Bug.upgrade("remove JDK version check once current adapter supports Cassandra 4.x");
+    boolean compatibleJdk = TestUtil.getJavaMajorVersion() < 11;
+    if (enabled && compatibleJdk) {
+      return ConditionEvaluationResult.enabled("Cassandra enabled");
```
Comment (vlsi) [inline cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java]: Is this still valid?
Does it probably support Java 11?

File: cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java (hunk 1 above)
Comment (asereda-gs) [reply cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java]: Well I'm not sure Cassandra was ever meant to be embedded ([example](https://github.com/apache/cassandra/blob/cassandra-3.11/src/java/org/apache/cassandra/service/CassandraDaemon.java#L577) from CassandraDaemon.java):

```java
// code from CassandraDaemon.java

public void stop() {
// ... 
// On windows, we need to stop the entire system as prunsrv doesn't have the jsvc hooks
// We rely on the shutdown hook to drain the node
  if (FBUtilities.isWindows)
         System.exit(0);
// .... 
}
```

`System.exit()` inside `Service.stop/close` is very telling. So one has to find work-arounds to correctly shutdown cassandra without stopping JVM

Comment (asereda-gs) [reply cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java]: We use [cassandra-unit](https://github.com/jsevellec/cassandra-unit) to manage cassandra unfortunately they don't expose `CassandraDaemon`.

Perhaps one can start Cassandra manually without cassandra-unit but I didn't want to introduce too many changes.

File: cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java (hunk 2 above)
Comment (asereda-gs) [reply cassandra/src/test/java/org/apache/calcite/test/CassandraExtension.java]: Cassandra support for JDK11 was introduced in 4.x (see [CASSANDRA-9608](https://issues.apache.org/jira/browse/CASSANDRA-9608) we still use 3.6

**Comment by DonnyZone** [comment]
@asereda-gs I checked this fix on two Windows OS machines. However, it still fails to delete the `.toDelete` file.  Any further information or operations I can do to help?

**Comment by asereda-gs** [comment]
By "fails to delete" do you mean throws exception or `.toDelete` folder is still there ?

**Comment by DonnyZone** [comment]
`.toDelete` folder is still there.

**Comment by DonnyZone** [comment]
Build log in machine 1:
```
> Task :cassandra:test

CassandraAdapterDataTypesTest STANDARD_OUT
    2020-02-13 09:30:02 [pool-2-thread-1] ERROR o.a.cassandra.service.StartupChecks - cassandra.jmx.local.port missing from cassandra-env.sh, unable to start local JMX service.

CassandraAdapterDataTypesTest STANDARD_ERROR
    no sigar-amd64-winnt.dll in java.library.path
    org.hyperic.sigar.SigarException: no sigar-amd64-winnt.dll in java.library.path
        at org.hyperic.sigar.Sigar.loadLibrary(Sigar.java:172)
        at org.hyperic.sigar.Sigar.<clinit>(Sigar.java:100)
        at org.apache.cassandra.utils.SigarLibrary.<init>(SigarLibrary.java:47)
        at org.apache.cassandra.utils.SigarLibrary.<clinit>(SigarLibrary.java:28)
        at org.apache.cassandra.service.StartupChecks$7.execute(StartupChecks.java:266)
        at org.apache.cassandra.service.StartupChecks.verify(StartupChecks.java:125)
        at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:200)
        at org.apache.cassandra.service.CassandraDaemon.activate(CassandraDaemon.java:602)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.lambda$startEmbeddedCassandra$1(EmbeddedCassandraServerHelper.java:150)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
          4.0sec, org.apache.calcite.test.CassandraAdapterDataTypesTest > testCollectionsRowType()
WARNING  40.0sec,   10 completed,   0 failed,   1 skipped, org.apache.calcite.test.CassandraAdapterDataTypesTest

..................................................................................
..................................................................................

> Task :cassandra:test
         32.6sec,   11 completed,   0 failed,   0 skipped, org.apache.calcite.test.CassandraAdapterTest

Gradle Test Executor 18 STANDARD_OUT
    2020-02-13 09:31:09 [ForkJoinPool-1-worker-1] ERROR o.a.cassandra.service.StorageService - Stopping gossiper
WARNING  83.1sec,   21 completed,   0 failed,   1 skipped, Gradle Test Run :cassandra:test
```

**Comment by DonnyZone** [comment]
Gradle build log in machine 2.
```
> Task :cassandra:test
FAILURE   0.0sec, org.apache.calcite.test.CassandraAdapterTest > initializationError

CassandraAdapterTest > initializationError FAILED
    org.junit.jupiter.api.extension.ParameterResolutionException: Failed to resolve parameter [com.datastax.driver.core.Session arg0] in method [static void org.apache.calcite.test.CassandraAdapterTest.load(com.datastax.driver.core.Session)]: All host(s) tried for query failed (tried: localhost/0:0:0:0:0:0:0:1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/0:0:0:0:0:0:0:1:9142] Cannot connect), localhost/127.0.0.1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/127.0.0.1:9142] Cannot connect))
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:239)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$8(ClassBasedTestDescriptor.java:371)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:369)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:193)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:77)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:141)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:121)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
        at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

        Caused by:
        com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/0:0:0:0:0:0:0:1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/0:0:0:0:0:0:0:1:9142] Cannot connect), localhost/127.0.0.1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/127.0.0.1:9142] Cannot connect))
            at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:268)
            at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:107)
            at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1652)
            at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1571)
            at com.datastax.driver.core.Cluster.init(Cluster.java:208)
            at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
            at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
            at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
            at org.cassandraunit.utils.EmbeddedCassandraServerHelper.initSession(EmbeddedCassandraServerHelper.java:231)
            at org.cassandraunit.utils.EmbeddedCassandraServerHelper.getSession(EmbeddedCassandraServerHelper.java:209)
            at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:136)
            at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:129)
            at org.apache.calcite.test.CassandraExtension.lambda$getOrCreate$0(CassandraExtension.java:101)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:84)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:88)
            at org.junit.jupiter.engine.execution.NamespaceAwareStore.getOrComputeIfAbsent(NamespaceAwareStore.java:61)
            at org.apache.calcite.test.CassandraExtension.getOrCreate(CassandraExtension.java:101)
            at org.apache.calcite.test.CassandraExtension.resolveParameter(CassandraExtension.java:77)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
            ... 33 more
FAILURE   8.6sec,    1 completed,   1 failed,   0 skipped, org.apache.calcite.test.CassandraAdapterTest

Gradle Test Executor 5 STANDARD_ERROR
    ▒▒▒▒ 13, 2020 9:19:04 ▒▒▒▒ org.junit.jupiter.engine.execution.JupiterEngineExecutionContext close
    ▒▒▒▒: Caught exception while closing extension context: org.junit.jupiter.engine.descriptor.JupiterEngineExtensionContext@3037b217
    java.lang.IllegalStateException: Can't use this cluster instance because it encountered an error in its initialization
        at com.datastax.driver.core.Cluster.checkNotClosed(Cluster.java:639)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:375)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
        at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.initSession(EmbeddedCassandraServerHelper.java:231)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.getSession(EmbeddedCassandraServerHelper.java:209)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:136)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:129)
        at org.apache.calcite.test.CassandraExtension.lambda$getOrCreate$0(CassandraExtension.java:101)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:58)
        at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:73)
        at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
        at org.junit.jupiter.engine.descriptor.JupiterEngineDescriptor.cleanUp(JupiterEngineDescriptor.java:67)
        at org.junit.jupiter.engine.descriptor.JupiterEngineDescriptor.cleanUp(JupiterEngineDescriptor.java:29)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$cleanUp$9(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.cleanUp(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:83)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
        at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
    Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/0:0:0:0:0:0:0:1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/0:0:0:0:0:0:0:1:9142] Cannot connect), localhost/127.0.0.1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/127.0.0.1:9142] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:268)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:107)
        at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1652)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1571)
        at com.datastax.driver.core.Cluster.init(Cluster.java:208)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
        at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.initSession(EmbeddedCassandraServerHelper.java:231)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.getSession(EmbeddedCassandraServerHelper.java:209)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:136)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:129)
        at org.apache.calcite.test.CassandraExtension.lambda$getOrCreate$0(CassandraExtension.java:101)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:84)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:88)
        at org.junit.jupiter.engine.execution.NamespaceAwareStore.getOrComputeIfAbsent(NamespaceAwareStore.java:61)
        at org.apache.calcite.test.CassandraExtension.getOrCreate(CassandraExtension.java:101)
        at org.apache.calcite.test.CassandraExtension.resolveParameter(CassandraExtension.java:77)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$8(ClassBasedTestDescriptor.java:371)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:369)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:193)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:77)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:141)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:121)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
        ... 6 more

FAILURE   0.0sec, UnknownClass > executionError

UnknownClass > executionError FAILED
    java.lang.IllegalStateException: Can't use this cluster instance because it encountered an error in its initialization
        at com.datastax.driver.core.Cluster.checkNotClosed(Cluster.java:639)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:375)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
        at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.initSession(EmbeddedCassandraServerHelper.java:231)
        at org.cassandraunit.utils.EmbeddedCassandraServerHelper.getSession(EmbeddedCassandraServerHelper.java:209)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:136)
        at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:129)
        at org.apache.calcite.test.CassandraExtension.lambda$getOrCreate$0(CassandraExtension.java:101)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
        at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:58)
        at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:73)
        at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
        at org.junit.jupiter.engine.descriptor.JupiterEngineDescriptor.cleanUp(JupiterEngineDescriptor.java:67)
        at org.junit.jupiter.engine.descriptor.JupiterEngineDescriptor.cleanUp(JupiterEngineDescriptor.java:29)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$cleanUp$9(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.cleanUp(NodeTestTask.java:151)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:83)
        at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
        at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

        Caused by:
        com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/0:0:0:0:0:0:0:1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/0:0:0:0:0:0:0:1:9142] Cannot connect), localhost/127.0.0.1:9142 (com.datastax.driver.core.exceptions.TransportException: [localhost/127.0.0.1:9142] Cannot connect))
            at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:268)
            at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:107)
            at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1652)
            at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1571)
            at com.datastax.driver.core.Cluster.init(Cluster.java:208)
            at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
            at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
            at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
            at org.cassandraunit.utils.EmbeddedCassandraServerHelper.initSession(EmbeddedCassandraServerHelper.java:231)
            at org.cassandraunit.utils.EmbeddedCassandraServerHelper.getSession(EmbeddedCassandraServerHelper.java:209)
            at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:136)
            at org.apache.calcite.test.CassandraExtension$CassandraResource.<init>(CassandraExtension.java:129)
            at org.apache.calcite.test.CassandraExtension.lambda$getOrCreate$0(CassandraExtension.java:101)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$getOrComputeIfAbsent$0(ExtensionValuesStore.java:81)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore$MemoizingSupplier.get(ExtensionValuesStore.java:182)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:84)
            at org.junit.jupiter.engine.execution.ExtensionValuesStore.getOrComputeIfAbsent(ExtensionValuesStore.java:88)
            at org.junit.jupiter.engine.execution.NamespaceAwareStore.getOrComputeIfAbsent(NamespaceAwareStore.java:61)
            at org.apache.calcite.test.CassandraExtension.getOrCreate(CassandraExtension.java:101)
            at org.apache.calcite.test.CassandraExtension.resolveParameter(CassandraExtension.java:77)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:183)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameters(ExecutableInvoker.java:144)
            at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:96)
            at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$8(ClassBasedTestDescriptor.java:371)
            at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
            at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:369)
            at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:193)
            at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:77)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
            at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
            at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
            at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
            at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
            at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:141)
            at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:121)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
            at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
            at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
            at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
            at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
            ... 6 more
FAILURE   0.0sec,    1 completed,   1 failed,   0 skipped, UnknownClass
FAILURE 103.6sec,    3 completed,   3 failed,   0 skipped, Gradle Test Run :cassandra:test

3 tests completed, 3 failed

> Task :cassandra:test FAILED
```

**Comment by asereda-gs** [comment]
For machine1 logs are fine (`SigarException` is expected). Can you check if `.toDelete` exists after running tests on machine1 ?

**Comment by DonnyZone** [comment]
I checked both of them, `.toDelete` file exists.

**Comment by DonnyZone** [comment]
snapshot

**Comment by asereda-gs** [comment]
Try now pls (refresh your local copy).
On machine1 where startup was successful.

**Comment by DonnyZone** [comment]
@asereda-gs Thanks! It works well on both machines.

**Pull request merged**